pkg/config/node.go

```diff
@@ -234,7 +234,9 @@ func IsUpgradeStillRunning(kubeconfigPath string) (error, bool) {
 	}
 
 	for _, node := range nodes.Items {
-		if node.Annotations["machineconfiguration.openshift.io/desiredConfig"] != node.Annotations["machineconfiguration.openshift.io/currentConfig"] {
+		if node.Annotations["machineconfiguration.openshift.io/desiredConfig"] != node.Annotations["machineconfiguration.openshift.io/currentConfig"] ||
+			node.Annotations["machineconfiguration.openshift.io/desiredConfig"] != nodes.Items[0].Annotations["machineconfiguration.openshift.io/desiredConfig"] {
```

Comment:
So this is essentially checking that the current node matches the first node. Do we know that the first node will always be upgraded first or should we make this check a separate loop where we verify all of the nodes' desiredConfigs match (or I guess we could still use this loop with a bit different logic)?

Reply:
A. No, we don't know that the first node will be first.

B. This code supposes to check that : 
    * desiredConfig is equal to currentConfig for each node 
and 
   * all the nodes has the same desiredConfig 

So, If  desiredConfig != currentConfig for one of the nodes, we'll fail because of the first condition in the if statement. 

And the second condition in the if statement verifies that all the nodes has the same  desiredConfig.

I used the below logic to verify that all nodes has the same desiredConfig.
Assuming that we have an array of N (0 - N-1) numbers and we need to check if all values of the array are equal.

So, if for each i between 1 to N-1 Arr[i] == Arr[0] that means that all array members are equal (because whenever x = y and y = z, then also x = z )

Let me know if you still think I should change/update something in this statement
.

Reply:
Oh, right. This does check that they're all equal. Oops. :-)